modules/nixpkgs.nix

```diff
@@ -1,26 +1,32 @@
-#
 # Nixpkgs module. The only exception to the rule.
-#
 # Provides a `pkgs` argument in `perSystem`.
-#
-# Arguably, this shouldn't be in flake-parts, but in nixpkgs.
-# Nixpkgs could define its own module that does this, which would be
-# a more consistent UX, but for now this will do.
-#
-# The existence of this module does not mean that other flakes' logic
-# will be accepted into flake-parts, because it's against the
-# spirit of Flakes.
-#
+topLevel@{ config, options, inputs, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib) mkPerSystemOption;
+
+in
 {
-  config = {
-    perSystem = { inputs', lib, ... }: {
-      config = {
-        _module.args.pkgs = lib.mkOptionDefault (
-          builtins.seq
-            (inputs'.nixpkgs or (throw "flake-parts: The flake does not have a `nixpkgs` input. Please add it, or set `perSystem._module.args.pkgs` yourself."))
-            inputs'.nixpkgs.legacyPackages
-        );
-      };
-    };
+  imports = [ inputs.nixpkgs.flakeModules.default ];
+
+  options = {
+    perSystem = mkPerSystemOption ({ config, system, ... }:
+      let
+        finalOverlay = lib.composeManyExtensions
+          # Last element has priority, so `perSystem` overlays rule
+          (topLevel.config.nixpkgs.overlays ++ config.nixpkgs.overlays);
+
+        finalPkgs = config.nixpkgs.mkPkgsFromArgs {
+          localSystem = { inherit system; };
+          overlays = [ finalOverlay ];
+        };
+
+      in
+      {
+        imports = [ inputs.nixpkgs.flakeModules.default ];
```

Comment:
As a general rule, modules are imported at the top level only. This makes importing more consistent, not having to wonder where to import it.
Maybe this was added by mistake? Maybe it doesn't even raise an error, as the laziness can mask problems sometimes; a blessing and a curse.

Reply:
I thought this would be needed to expose the `nixpkgs` option to the `perSystem` evaluation scope. Will experiment with removing it now.

Reply:
> I see you've avoided the perSystem option there. I think users would expect the Nixpkgs to be in perSystem.

See my comment above. Am I wrong? I'm using `nixpkgs` option inside the `perSystem` scope [here](https://github.com/sajban/clojix/blob/f1e58a7fee1ebcdc0dee82781d1ca81c95a9cfd1/nix/packagesModule.nix#L94)